Proofs/SHA512/SHA512_block_armv8_rules.lean

```diff
@@ -134,17 +134,25 @@ theorem concat_of_rsh_is_msb_128 (x y : BitVec 64) :
   (x ++ y) >>> 64 = BitVec.zeroExtend 128 x := by
   bv_check "lrat_files/Sha512_block_armv8_rules.lean-concat_of_rsh_is_msb_128-101-2.lrat"
 
--- (FIXME) Generalize to arbitrary-length bitvecs.
 theorem truncate_of_concat_is_lsb_64 (x y : BitVec 64) :
   BitVec.zeroExtend 64 (x ++ y) = y := by
   bv_check "lrat_files/Sha512_block_armv8_rules.lean-truncate_of_concat_is_lsb_64-106-2.lrat"
 
--- (FIXME) Generalize to arbitrary-length bitvecs.
+theorem truncate_of_concat_is_lsb (w : Nat) (x y : BitVec w) :
+  BitVec.zeroExtend w (x ++ y) = y := by
+  ext
+  simp [BitVec.leftshift_n_or_mod_2n]
+
 theorem zeroextend_bigger_smaller_64 (x : BitVec 64) :
   BitVec.zeroExtend 64 (BitVec.zeroExtend 128 x) =
   BitVec.zeroExtend 64 x := by
   bv_omega
 
+theorem zeroextend_bigger_smaller (w : Nat) (x : BitVec w) :
```

Comment:
This could be generalized further. In fact, the lemma we really need already exists in Lean --  `BitVec.zeroExtend_zeroExtend_of_le`. We should really delete `zeroextend_bigger_smaller_64` and `zeroextend_bigger_smaller`, and use that lemma instead.